tests/security/ConcurrentAuthLimit/ConcurrentAuthLimit.cpp

```diff
@@ -0,0 +1,386 @@
+#include <dds/DCPS/RTPS/BaseMessageTypes.h>
+#include <dds/DCPS/RTPS/GuidGenerator.h>
+#include <dds/DCPS/RTPS/MessageTypes.h>
+#include <dds/DCPS/RTPS/ParameterListConverter.h>
+#include <dds/DCPS/RTPS/RtpsDiscovery.h>
+
+#include <dds/DCPS/security/framework/Properties.h>
+#include <dds/DCPS/security/AuthenticationBuiltInImpl.h>
+
+#include <dds/DCPS/transport/rtps_udp/RtpsUdpInst.h>
+#ifdef ACE_AS_STATIC_LIBS
+#include <dds/DCPS/transport/rtps_udp/RtpsUdp.h>
+#endif
+
+#include <dds/DCPS/LogAddr.h>
+
+#include <ace/SOCK_Dgram.h>
+#include <ace/SOCK_Dgram_Mcast.h>
+
+using namespace OpenDDS::DCPS;
+using namespace OpenDDS::RTPS;
+
+void append(DDS::PropertySeq& props, const char* name, const OpenDDS::DCPS::String& value, bool propagate = false)
+{
+  const DDS::Property_t prop = {name, value.c_str(), propagate};
+  const unsigned int len = props.length();
+  props.length(len + 1);
+  props[len] = prop;
+}
+
+bool receive(const OpenDDS::DCPS::GUID_t& participant_guid,
+             ACE_SOCK_Dgram_Mcast& multicast_socket)
+{
+  // Wait for an SPDP announcement before proceeding.
+  ACE_Message_Block buff(64 * 1024);
+
+  ACE_INET_Addr remote;
+  buff.reset();
+
+#ifdef ACE_LACKS_SENDMSG
+  const ssize_t bytes = multicast_socket.recv(buff.wr_ptr(), buff.space(), remote);
+#else
```

Comment:
Is there any advantage to having both the simple recv and the iovec recv alternatives?